src/wp-includes/media.php

```diff
@@ -1723,25 +1736,57 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 			$value = 'lazy';
 		}
 
-		// Images should have dimension attributes for the `loading` attribute to be added.
-		if ( false === strpos( $image, ' width=' ) || false === strpos( $image, ' height=' ) ) {
+		// Images should have source and dimension attributes for the `loading` attribute to be added.
+		if ( false === strpos( $image, ' src=' ) || false === strpos( $image, ' width=' ) || false === strpos( $image, ' height=' ) ) {
```

Comment:
We are looking for `'/src="([^"]+)"/'` in wp_img_tag_add_width_and_height_attr(). Thinking a close match would be to look for ` src="` here. It is possible to have ` src=` in an alt attribute, for example; less likely for it to be ` src="`.

The general logic is that it's ok to have a small chance for "false positives" with these check, i.e. err on not adding the attribute, but be 100% sure we're not adding a second one.

Reply:
Actually, nevermind. It's good as-is :)

Reply:
Hmm, or maybe not... See https://core.trac.wordpress.org/ticket/44427#comment:100.